### ISSUE TITLE
invalid recursive type itemCanvasObject

### ISSUE THREAD

**Issue by Epenance** [fyne-io/fyne#5005]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When trying to either build a completely new project, or run the fyne_demo I get the following errors:

```
# fyne.io/fyne/v2/internal/async
../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:28:6: invalid recursive type itemCanvasObject
        ../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:28:6: itemCanvasObject refers to
        ../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:29:13: atomic.Pointer refers to
        ../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:28:6: itemCanvasObject
../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:40:4: i.next undefined (type *itemCanvasObject has no field or method next)
../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:41:4: i.v undefined (type *itemCanvasObject has no field or method v)
../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:46:19: last.next undefined (type *itemCanvasObject has no field or method next)
../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:49:13: last.next undefined (type *itemCanvasObject has no field or method next)
../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:68:21: first.next undefined (type *itemCanvasObject has no field or method next)
../../go/pkg/mod/fyne.io/fyne/v2@v2.5.0/internal/async/queue_canvasobject.go:77:20: firstnext.v undefined (type *itemCanvasObject has no field or method v)

```

This seems to only be the case on 2.5.0 as downgrading to 2.4.5 works.

### How to reproduce

1.  go run fyne.io/fyne/v2/cmd/fyne_demo@latest
2. See errors

### Example code

package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Hello World")

	w.SetContent(widget.NewLabel("Hello World!"))
	w.ShowAndRun()
}

### Fyne version

2.5.0

### Go compiler version

1.19.4

### Operating system and version

macOS Sonoma 14.5

### Additional Information

Not sure if this is only the case on MacOS as I haven't been able to test it on a Windows PC

**Comment by andydotxyz** [comment]
It looks like (for some compilers only?) the move from `*itemCanvasObject` to `atomic.Pointer[itemCanvasObject]` is triggering compile failures... any idea @Jacalz ?

Can you confirm where you got the Go install from @Epenance ?

**Comment by Epenance** [comment]
Simply followed the instructions here to try and set it up on my macbook:

https://docs.fyne.io/started/

Also tried letting Goland install it for me with the same results.

**Comment by Jacalz** [comment]
This seems very strange. Works fine on my machine. Can you try upgrading to something newer than Go 1.19?

**Comment by Epenance** [comment]
Upgrading to 1.22.5 worked for me

**Comment by Jacalz** [comment]
Is it solved by 1.20, 1.21 or 1.22 though? If the issue is 1.19 specific, we might need to fix it for that version.

**Comment by Jacalz** [comment]
Hmm, actually. I wonder if this was bug that perhaps was fixed in the Go 1.19 maintenance window. You mention using 1.19.4 but that's definitely not the latest version in that release series. Would you mind trying 1.19 again but with the latest bug fix release?

**Comment by Epenance** [comment]
Looks like you are right, tried to change to version to a flat 1.19 and it seems to work as well :) Thanks for the quick replies
